crates/hir-expand/src/lib.rs

```diff
@@ -1091,3 +1050,76 @@ impl ExpandTo {
 }
 
 intern::impl_internable!(ModPath, attrs::AttrInput);
+
+#[salsa::interned(no_lifetime)]
+pub struct MacroCallId {
+    pub loc: MacroCallLoc,
+}
+
+impl From<span::MacroCallId> for MacroCallId {
+    #[inline]
+    fn from(value: span::MacroCallId) -> Self {
+        MacroCallId::from_id(value.0)
+    }
+}
+
+impl From<MacroCallId> for span::MacroCallId {
+    #[inline]
+    fn from(value: MacroCallId) -> span::MacroCallId {
+        span::MacroCallId(value.as_id())
+    }
+}
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, salsa::Supertype)]
+pub enum HirFileId {
+    FileId(EditionedFileId),
+    MacroFile(MacroCallId),
+}
```

Comment:
Note that it would be nicei f we could shrink this to 4 bytes again (as was done with the bit packing before). Iirc back when I implemented that it saved ~10mb. Not relevant to this PR for now though.

Reply:
We can dot this if we store it in places as `salsa::Id` (or some newtype wrapper). Methods can still take `HirFileId`.

Reply:
Yes, I believe Niko's initial supertype proposal wouldvebeen something like that, where instead of it being a derive it was an attribute that collapsed the variants into an erased wrapper. I wonder if we could implement that as a separate thing to the derive as well. I do like the derive in most cases, given it doesn't require you to extract the active variant manually.